### ISSUE TITLE
Sporadic Spec Failure: uploads_request_spec

### ISSUE THREAD

**Issue by omgitsbillryan** [department-of-veterans-affairs/va.gov-team#7305]
## The Problem
The following spec has failed the `master` vets-api build a number of times, and therefore prevented the daily deploy (2 times so far that I counted, [1](https://dsva.slack.com/archives/C0MQ281DJ/p1585247667002600), [2](https://dsva.slack.com/archives/C0MQ281DJ/p1584640829086900)).

```
01:56:19    1) VBA Document Uploads Endpoint #download /v1/uploads/{id} returns a 200 with content-type of zip
01:56:19       Failure/Error: expect(response.status).to eq(200)
01:56:19  
01:56:19         expected: 200
01:56:19              got: 500
01:56:19  
01:56:19         (compared using ==)
01:56:19       # ./modules/vba_documents/spec/request/v1/uploads_request_spec.rb:139:in `block (3 levels) in <top (required)>'
```

## Additional Info
The spec mostly seems to pass but will sporadically fail.

## Work to be Done
One of:
- [ ] Discover why it's flapping & fix
or
- [ ] Remove spec altogether & write new examples if coverage is lacking

**Comment by omgitsbillryan** [comment]
Adding a data point - this failed a `master` build here: http://jenkins.vfs.va.gov/job/testing/job/vets-api/job/master/2741/console

which caused a failed daily deploy here: https://dsva.slack.com/archives/C0MQ281DJ/p1591726234246200

**Comment by omgitsbillryan** [comment]
New data point coming in hot: https://dsva.slack.com/archives/C0MQ281DJ/p1592589632285100

**Comment by wyattwalter** [comment]
@omgitsbillryan this is a Lighthouse-managed module, right? Can we ask someone there to work on it? They may or may not have some more time to work on this.

**Comment by f1337** [comment]
@wyattwalter @omgitsbillryan I should have some time to look into it this afternoon. I'm coming in completely green/new to the issue, and am open to any input. Bill shared the back story and a bunch of context in Slack.

**Comment by johnpaulashenfelter** [comment]
Temporarily handled by https://github.com/department-of-veterans-affairs/vets-api/pull/4439

**Comment by f1337** [comment]
I ran the specs locally, using the seed mentioned, and… cannot reproduce. I'm still digging.

**Comment by omgitsbillryan** [comment]
Here's the trace:
```
14:08:27    1) VBA Document Uploads Endpoint #download /v1/uploads/{id} returns a 200 with content-type of zip
14:08:27       Failure/Error: expect(response.status).to eq(200)
14:08:27         Expected 500 to eq 200.
14:08:27       # /usr/local/bundle/gems/super_diff-0.4.2/lib/super_diff/rspec/monkey_patches.rb:40:in `handle_failure'
14:08:27       # ./modules/vba_documents/spec/request/v1/uploads_request_spec.rb:139:in `block (3 levels) in <top (required)>'
```

**Comment by omgitsbillryan** [comment]
Oh woops - that trace ☝️ is already in the OP. I looked through http://jenkins.vfs.va.gov/job/testing/job/vets-api/job/master/2741/consoleFull and was unable to find the _full_ trace. I don't think our build spits it out.

**Comment by f1337** [comment]
> Oh woops - that trace ☝️ is already in the OP. I looked through http://jenkins.vfs.va.gov/job/testing/job/vets-api/job/master/2741/consoleFull and was unable to find the _full_ trace. I don't think our build spits it out.

Ah, OK, thank you @omgitsbillryan!

**Comment by f1337** [comment]
I found a potentially related issue in the [vets-api_job_master_2741.log](https://github.com/department-of-veterans-affairs/va.gov-team/files/4852830/vets-api_job_master_2741.log) (thank you @kfrz for grabbing it for me!):

```
[2020-06-09T18:08:25.402Z] [error] Breaking on: "x\x01\xE2\x80\xA6Q\xC2\xBDn\xC3..." at offset 0x59
[2020-06-09T18:08:25.402Z] [error] Last exception: [Origami::InvalidObjectError] Failed to parse object (no:4,gen:0)
[2020-06-09T18:08:25.402Z]       -> [Origami::InvalidStreamObjectError] Stream shall end with a 'endstream' statement
[2020-06-09T18:08:25.402Z] [error] Cannot read : "x\x01\xE2\x80\xA6Q\xC2\xBDn\xC3..."
[2020-06-09T18:08:25.402Z] [error] Stopped on exception : Failed to parse object (no:4,gen:0)
[2020-06-09T18:08:25.402Z]       -> [Origami::InvalidStreamObjectError] Stream shall end with a 'endstream' statement
```

**Comment by f1337** [comment]
We have two viable options for remediation while we investigate root cause. I suggest we need to pick one of the following, merge it, and open a new ticket for root-cause determination.

* Skip the specs: https://github.com/department-of-veterans-affairs/vets-api/pull/4453
* Retry on fail: https://github.com/department-of-veterans-affairs/vets-api/pull/4464

I'm also curious if we've seen this issue reappear since merging the PR which removes [iconv](https://github.com/department-of-veterans-affairs/vets-api/pull/4460)? The underlying error message I found (in a prior comment above), leads me to think the two issues may be related.

**Comment by omgitsbillryan** [comment]
+1 for using retry.

Once retry is in place, I would also then be in favor of just dropping this work. Root causing these "order of running specs" bugs is hard enough, but then consider that we're not the original author of this spec nor do we have great expertise on the feature itself. Worst case scenario, the spec re-runs once every so often and adds some fraction of a second to the build time.

**Comment by f1337** [comment]
I agree with @omgitsbillryan: +1 for punting after this, *especially* if we don't hear any more about it. I still suspect the iconv bug was at the root (it was too coincidental for my taste), but _proving_ that will be a massive time sink. Not worth the effort to continue IMO.

**Comment by alexpappasoddball** [comment]
Closing this issue per discussion with @f1337 and @omgitsbillryan
